sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -190,40 +229,53 @@ public static BinaryData fromObject(Object data) {
     }
 
     /**
-     * Serialize the given {@link Object} into {@link BinaryData} using json serializer which is available on classpath.
-     * The serializer on classpath must implement {@link JsonSerializer} interface. If the given Object is {@code null},
-     * an empty {@link BinaryData} will be returned.
-     * <p><strong>Code sample</strong></p>
-     * {@codesnippet com.azure.core.util.BinaryData.fromObjectAsync}
-
-     * @param data The object to use as data backing the instance of {@link BinaryData}.
-     * @throws IllegalStateException If a {@link JsonSerializer} cannot be found on the classpath.
-     * @return {@link BinaryData} representing the JSON serialized object.
+     * Creates an instance of {@link BinaryData} by serializing the {@link Object} using the default {@link
+     * JsonSerializer}.
+     * <p>
+     * If {@code data} is null an empty {@link BinaryData} will be returned.
+     * <p>
+     * <b>Note:</b> A {@link JsonSerializer} implementation must be available on the classpath.
+     *
+     * <p><strong>Creating an instance from an Object</strong></p>
+     *
+     * {@codesnippet com.azure.core.util.BinaryData.fromObjectAsync#Object}
      *
+     * @param data The object that will be JSON serialized that {@link BinaryData} will represent.
+     * @return A {@link Mono} of {@link BinaryData} representing the JSON serialized object.
+     * @throws IllegalStateException If a {@link JsonSerializer} implementation cannot be found on the classpath.
      * @see JsonSerializer
-     * @see <a href="ObjectSerializer" target="_blank">More about serialization</a>
      */
     public static Mono<BinaryData> fromObjectAsync(Object data) {
-        return Mono.fromCallable(() -> fromObject(data));
+        try {
+            return Mono.fromCallable(() -> fromObject(data));
+        } catch (RuntimeException runtimeException) {
+            return Mono.error(runtimeException);
+        }
```

Comment:
I don't think this try-catch is required. Any exception thrown from `fromObject` method will be converted to an error signal in reactor.